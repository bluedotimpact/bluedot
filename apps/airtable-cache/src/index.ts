import { getInstance } from './app';
import { migrateDb } from './db/migrations/migrator';
import { startCronJobs } from './lib/cron';

const start = async () => {
  try {
    console.log('Server starting...');
    await migrateDb();
    startCronJobs();
    const instance = await getInstance();
    await instance.listen({
      // eslint-disable-next-line turbo/no-undeclared-env-vars
      port: process.env.PORT ? parseInt(process.env.PORT) : 8001,
      host: '0.0.0.0',
    }).then((address) => {
      console.log(`Server listening on ${address}`);
    });
  } catch (err) {
    console.error('Failed to start server', err);
    process.exit(1);
  }
};

start();
