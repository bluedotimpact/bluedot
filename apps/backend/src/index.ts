import { app } from './app';
import { migrateDb } from './db/migrations/migrator';

const start = async () => {
  try {
    await migrateDb();
    await app.listen({ port: 8001 });
  } catch (err) {
    console.error(err);
    process.exit(1);
  }
};

start();
