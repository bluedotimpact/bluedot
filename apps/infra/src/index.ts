import './vultr';
import './k8s';
import './minio';

export { k8sConfig } from './vultr/vke';
