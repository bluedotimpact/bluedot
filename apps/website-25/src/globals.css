/* Try to avoid adding styles here. Instead, use tailwind inline styles or add them to @bluedot/ui's shared.css (if they're relevant to other apps) */
@tailwind base;
@tailwind components;
@tailwind utilities;

@layer base {
  h1 {
    @apply text-color-secondary-text font-sans text-size-xl font-bold leading-tight;
  }
  h2, .title {
    @apply text-color-secondary-text font-sans text-size-xl font-bold leading-normal;
  }
  h3, .subtitle {
    @apply text-color-secondary-text font-sans text-size-l font-[650] leading-normal;
  }
  h4 {
    @apply text-color-secondary-text font-sans text-xl font-[650];
  }
  p, li, .body {
    @apply text-color-text font-sans text-size-m font-normal leading-normal;
  }
  ul {
    @apply list-inside list-disc;
  }
  ol {
    @apply list-decimal;
  }
  .text-on-dark {
    @apply text-color-text-on-dark;
  }
  a, .link {
    @apply text-color-secondary-text hover:text-color-primary-accent hover:cursor-pointer;
  }
  .link-on-dark {
    @apply text-color-text-on-dark hover:text-color-secondary-accent hover:cursor-pointer;
  }
}
