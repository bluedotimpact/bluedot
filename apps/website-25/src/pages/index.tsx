import {
  HeroSection,
} from '@bluedot/ui';
import CourseSection from '../components/homepage/CourseSection';
import GraduateSection from '../components/homepage/GraduateSection';
import StorySection from '../components/homepage/StorySection';
import CommunitySection from '../components/homepage/CommunitySection/index';

const HomePage = () => {
  return (
    <div>
      <HeroSection
        title="The expertise you need to shape safe AI "
        subtitle="We run the world's most trusted AI Safety educational courses, career services and support community. Our programs are developed in collaboration with AI Safety world experts."
      >
        <div className="hero-section__logo-container flex flex-col items-center gap-7 mb-3">
          <img className="hero-section__logo-icon w-20 mb-20" src="/images/logo/BlueDot_Impact_Icon_White.svg" alt="BlueDot Impact" />
        </div>
      </HeroSection>
      <GraduateSection />
      <CourseSection />
      <StorySection />
      <CommunitySection />
    </div>
  );
};

export default HomePage;
