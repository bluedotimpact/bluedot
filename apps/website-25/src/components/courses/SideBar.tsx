import React from 'react';
import { CourseUnit } from '@bluedot/ui/src/constants';

type SideBarProps = {
  // Required
  units: CourseUnit[];
  currentUnit: CourseUnit;
};

const SideBar: React.FC<SideBarProps> = ({
  units,
  currentUnit,
}) => {
  const isCurrentUnit = (unit: CourseUnit) => {
    return currentUnit.title === unit.title;
  };
  return (
    <div className="sidebar w-full md:w-[332px] flex flex-col">
      <div className="sidebar__content flex flex-col gap-6">
        {units.map((unit) => (
          <div
            className={`sidebar__unit border ${isCurrentUnit(unit) ? 'border-color-primary' : 'border-color-divider'} rounded-radius-md p-4 flex flex-col gap-2`}
          >
            <div className="sidebar__unit-item flex flex-col gap-2">
              <p className="sidebar__title uppercase text-size-xs font-bold">{unit.title}</p>
              <a href={unit.href} className="sidebar__description subtitle-sm">{unit.description}</a>
            </div>
            {isCurrentUnit(unit) && unit.chapters && (
              <div className="sidebar__chapters flex flex-col gap-4 border-t-2 border-color-primary pt-4">
                {unit.chapters.map((chapter) => (
                  <div className="sidebar__chapter">
                    <p className="sidebar__chapter-detail text-size-xs">
                      <span className="sidebar__chapter-type font-bold text-color-secondary-text">{chapter.type}:</span> <span className="sidebar__chapter-title">{chapter.title}</span>
                    </p>
                  </div>
                ))}
              </div>
            )}
          </div>
        ))}
      </div>
    </div>
  );
};

export default SideBar;
