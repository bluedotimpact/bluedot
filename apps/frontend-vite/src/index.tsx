import React from 'react';
import ReactDOM from 'react-dom/client';
import { createBrowserRouter, RouterProvider } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import './index.css';
import { HomePage } from './pages/HomePage';
import { RouteErrorPage } from './pages/ErrorPage';
import { AuthedPage } from './pages/AuthedPage';
import { LoginOauthCallbackPage, LoginPage } from './pages/LoginPage';

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: 0,
    },
  },
});

const router = createBrowserRouter([
  {
    path: '/',
    element: <HomePage />,
    errorElement: <RouteErrorPage />,
  },
  {
    path: '/login',
    element: <LoginPage />,
  },
  {
    path: '/login/oauth-callback',
    element: <LoginOauthCallbackPage />,
  },
  {
    path: '/authed',
    element: <AuthedPage />,
  },
]);

ReactDOM.createRoot(document.getElementById('root')!).render(
  <React.StrictMode>
    <QueryClientProvider client={queryClient}>
      <RouterProvider router={router} />
    </QueryClientProvider>
  </React.StrictMode>,
);
