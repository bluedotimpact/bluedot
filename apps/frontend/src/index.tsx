import React from 'react';
import ReactDOM from 'react-dom/client';
import { createBrowserRouter, RouterProvider } from 'react-router-dom';
import './index.css';
import { HomePage } from './pages/HomePage';
import { RouteErrorPage } from './pages/ErrorPage';
import { AuthedPage } from './pages/AuthedPage';
import { LoginOauthCallbackPage, LoginPage } from './pages/LoginPage';

const router = createBrowserRouter([
  {
    path: '/',
    element: <HomePage />,
    errorElement: <RouteErrorPage />,
  },
  {
    path: '/login',
    element: <LoginPage />,
  },
  {
    path: '/login/oauth-callback',
    element: <LoginOauthCallbackPage />,
  },
  {
    path: '/authed',
    element: <AuthedPage />,
  },
]);

ReactDOM.createRoot(document.getElementById('root')!).render(
  <React.StrictMode>
    <RouterProvider router={router} />
  </React.StrictMode>,
);
