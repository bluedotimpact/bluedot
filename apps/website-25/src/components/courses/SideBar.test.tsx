import { render } from '@testing-library/react';
import { describe, expect, test } from 'vitest';
import { COURSE_UNITS } from '@bluedot/ui/src/constants';
import SideBar from './SideBar';

describe('SideBar', () => {
  test('renders default as expected', () => {
    const { container } = render(<SideBar units={COURSE_UNITS} currentUnit={COURSE_UNITS[0]} />);
    expect(container).toMatchSnapshot();
  });
});
