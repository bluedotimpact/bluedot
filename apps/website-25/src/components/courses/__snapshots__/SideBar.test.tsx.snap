// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SideBar > renders default as expected 1`] = `
<div>
  <div
    class="sidebar w-full md:w-[332px] flex flex-col"
  >
    <div
      class="sidebar__content flex flex-col gap-6"
    >
      <div
        class="sidebar__unit border border-color-primary rounded-radius-md p-4 flex flex-col gap-2"
      >
        <div
          class="sidebar__unit-item flex flex-col gap-2"
        >
          <p
            class="sidebar__title uppercase text-size-xs font-bold"
          >
            Unit 1
          </p>
          <a
            class="sidebar__description subtitle-sm"
            href="/courses/future-of-ai/units/1"
          >
            Beyond chatbots: the expanding frontier of AI capabilities
          </a>
        </div>
        <div
          class="sidebar__chapters flex flex-col gap-4 border-t-2 border-color-primary pt-4"
        >
          <div
            class="sidebar__chapter"
          >
            <p
              class="sidebar__chapter-detail text-size-xs"
            >
              <span
                class="sidebar__chapter-type font-bold text-color-secondary-text"
              >
                Reading
                :
              </span>
               
              <span
                class="sidebar__chapter-title"
              >
                How current AI systems work
              </span>
            </p>
          </div>
          <div
            class="sidebar__chapter"
          >
            <p
              class="sidebar__chapter-detail text-size-xs"
            >
              <span
                class="sidebar__chapter-type font-bold text-color-secondary-text"
              >
                Demo
                :
              </span>
               
              <span
                class="sidebar__chapter-title"
              >
                They can build things
              </span>
            </p>
          </div>
          <div
            class="sidebar__chapter"
          >
            <p
              class="sidebar__chapter-detail text-size-xs"
            >
              <span
                class="sidebar__chapter-type font-bold text-color-secondary-text"
              >
                Reading
                :
              </span>
               
              <span
                class="sidebar__chapter-title"
              >
                The speed of progress
              </span>
            </p>
          </div>
          <div
            class="sidebar__chapter"
          >
            <p
              class="sidebar__chapter-detail text-size-xs"
            >
              <span
                class="sidebar__chapter-type font-bold text-color-secondary-text"
              >
                Reading
                :
              </span>
               
              <span
                class="sidebar__chapter-title"
              >
                From tools to agents
              </span>
            </p>
          </div>
          <div
            class="sidebar__chapter"
          >
            <p
              class="sidebar__chapter-detail text-size-xs"
            >
              <span
                class="sidebar__chapter-type font-bold text-color-secondary-text"
              >
                Exercise
                :
              </span>
               
              <span
                class="sidebar__chapter-title"
              >
                5 years ago, 5 years ahead
              </span>
            </p>
          </div>
        </div>
      </div>
      <div
        class="sidebar__unit border border-color-divider rounded-radius-md p-4 flex flex-col gap-2"
      >
        <div
          class="sidebar__unit-item flex flex-col gap-2"
        >
          <p
            class="sidebar__title uppercase text-size-xs font-bold"
          >
            Unit 2
          </p>
          <a
            class="sidebar__description subtitle-sm"
            href="/courses/future-of-ai/units/2"
          >
            Artificial general intelligence: on the horizon?
          </a>
        </div>
      </div>
      <div
        class="sidebar__unit border border-color-divider rounded-radius-md p-4 flex flex-col gap-2"
      >
        <div
          class="sidebar__unit-item flex flex-col gap-2"
        >
          <p
            class="sidebar__title uppercase text-size-xs font-bold"
          >
            Unit 3
          </p>
          <a
            class="sidebar__description subtitle-sm"
            href="/courses/future-of-ai/units/3"
          >
            AGI will drastically change how we live
          </a>
        </div>
      </div>
      <div
        class="sidebar__unit border border-color-divider rounded-radius-md p-4 flex flex-col gap-2"
      >
        <div
          class="sidebar__unit-item flex flex-col gap-2"
        >
          <p
            class="sidebar__title uppercase text-size-xs font-bold"
          >
            Unit 4
          </p>
          <a
            class="sidebar__description subtitle-sm"
            href="/courses/future-of-ai/units/4"
          >
            What can be done?
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
